standalone-tools/wss-scan.sh

```diff
@@ -0,0 +1,121 @@
+###### Information ############################################################################
+# Name:          wss-scan.sh
+# Maintainer:    ODFE Infra Team
+# Language:      Shell
+#
+# About:         This script is to scan the odfe distros for vulnerabilities and licenses
+#                It will scan the repositories and send the WhiteSource link to the mail 
+#                of the user. 
+#
+# Prerequisites: Need to set the Java Path
+#                Need to set the recepient mail in wss-scan.config
+#
+# Usage:         ./wss-scan.sh
+#
+# Starting Date: 10-13-2020
+# Modified Date: 10-15-2020
+###############################################################################################
+
+#Download the WhiteSource Agent 
+curl -LJO -sS https://github.com/whitesource/unified-agent-distribution/releases/latest/download/wss-unified-agent.jar > /dev/null
+
+echo "Download completed"
+
+#Scanning the config file for the user configurations
+source wss-scan.config
+gitRepos=${gitRepos//,/$'\n'}  # change the semicolons to white space
+
+basepath=$baseDirPath"/repos"
+
+if [[ ! -e $basepath ]]; then
+    mkdir -p $basepath
+fi
+ 
+
+echo "Cleaning up scan directories if already present"
+rm -rf $basepath/*
+
+
+#Cloning the desired Repos for scanning 
+for repo in $gitRepos
+do
+        echo "Cloning repo "$gitBasePath$repo
+    git clone "$gitBasePath$repo".git $basepath"/"$repo
+done
+
+cp /dev/null info.txt
+
+#Scanning the Repos using the WhiteSource Unified Agent
+for repo in $gitRepos
+do
+	if [ -d "$basepath"/"$repo" ]
+        then
+           echo "Scanning repo :"$gitBasePath$repo " Project:" $repo 
+	   repo_path=$basepath"/"$repo
+	   java -jar wss-unified-agent.jar -c wss-unified-agent.config -d $repo_path -apiKey $wss_apikey -product ODFE -project $repo | grep "Project name" | sed 's/^.\{,41\}//' >> info.txt 2>&1 &
+	else
+	   echo "Scanning failed for repo :"$gitBasePath$repo " Project:" $repo
+        fi
+done
+
+
+
+#Waiting for the scannings to complete
```

Comment:
Are you running the scans for all the repos, then waiting for all of them to complete?
The issue with this approach is, how do you know one of this process is not going to stuck/zombied in the bg and the script never get completed?

Can we have some hard limit here in case that thing happens?

Reply:
Set a limit to break the loop after 70 mins. The scheduled job is taking around 50 mins to complete for all the repos.